autopush-common/src/reliability.rs

```diff
@@ -239,26 +332,67 @@ impl PushReliability {
                     // Write the expiration only if the state is non-terminal. Otherwise we run the risk of
                     // messages reporting a false "expired" state even if they were "successful".
                     let key = ExpiryKey {
-                        id: id.to_owned(),
+                        id: id.to_string(),
                         state: new.to_owned(),
                     }
                     .to_string();
-                    let _ = pipe.zadd(EXPIRY, &key, expr.unwrap_or_default());
+                    pipe.zadd(EXPIRY, &key, expr.unwrap_or_default());
                     trace!("🔍 internal record result: {:?}", key);
                 }
+                trace!("🔍 upping {:?}", &new);
+                // Bump up the new state count, and set the state key's state if it still exists.
+                pipe.hincr(COUNTS, new.to_string(), 1);
+                let options = redis::SetOptions::default()
+                    .conditional_set(redis::ExistenceCheck::XX)
```

Comment:
otherwise this set will reset the TTL:
```suggestion
                    .with_expiration(redis::SetExpiry::KEEPTTL)
                    .conditional_set(redis::ExistenceCheck::XX)
```

Reply:
Forgot this